socorro/external/postgresql/base.py

```diff
@@ -111,25 +97,34 @@ def count(self, sql, params=None, error_message=None, connection=None):
                       will be opened.
 
         """
+        return self._execute(
+            single_value_sql,
+            sql,
+            error_message or "Failed to execute count against PostgreSQL",
+            params=params,
+            connection=connection
+        )
+
+    def _execute(
+        self, transformer, sql, error_message, params=None, connection=None
+    ):
         fresh_connection = False
         try:
             if not connection:
                 connection = self.database.connection()
                 fresh_connection = True
             # self.context.logger.debug(connection.cursor.mogrify(sql, params))
-            result = single_value_sql(connection, sql, params)
+            result = transformer(connection, sql, params)
```

Comment:
I do not understand the name of this method, ```transformer```.  Perhaps, something like ```transaction_function``` would be clearer.